### PR TITLE
UniV1+erc777重入攻击实验完成！

### PR DESCRIPTION
**Pull request by arc0035**

### CODE REVIEW

**Comment by arc0035** [comment]
@yanyanho

File: security/1-reentrancy/uniAndImbtc/README.md [hunk 1]
```diff
@@ -1,15 +1,163 @@
...
+## Basics of uniswap: constant K principle
+To deeper understand how the attack works, let's recap the basics of uniswap.
+
+In theory, when we exchange ERC777 for ether, the amount of ether bought can be conducted following the rule of constant K:
+
+![](1.png)
+
+And here is the code:
+
+```
```
Comment (0x-stan) [inline security/1-reentrancy/uniAndImbtc/README.md:45]: give highlight with add solidity after dots
```solidity

File: security/1-reentrancy/uniAndImbtc/README.md [hunk 2]
```diff
@@ -1,15 +1,163 @@
...
+It is possible to conduct a reentrancy attack on an UniswapV1 exchange hosting ERC777-Eth pair.To do this , create an attacker contract contains the following methods:
+
+- prepare: approve attacker to operate its ERC777 assets && adding hooks on ERC1820
+- trigger: a payable function that :
+  - receives some ether
+  - buy  ERC777 through uniswap
+  - sell a fraction of ERC777  to ether through uniswap exchange that will transfer ether to attacker contract and take ERC777 from attacker contract to uniswap exhange
+  
+- tokenToSend: This is receiver function called by ERC777. It will sell a fraction of ERC777 to ether through uniswap exchange. 
+-
```
Comment (0x-stan) [inline security/1-reentrancy/uniAndImbtc/README.md:22]: May remove blank line

Comment (arc0035) [reply security/1-reentrancy/uniAndImbtc/README.md:22]: Yes sir

File: security/1-reentrancy/uniAndImbtc/README.md (hunk 1 above)
Comment (arc0035) [reply security/1-reentrancy/uniAndImbtc/README.md:45]: Yes Sir

**Review by 0x-stan — APPROVED** [review]

**Comment by arc0035** [comment]
@yingjingyang

**Review by yingjingyang — APPROVED** [review]

**Pull request merged**